src/odemis/driver/technolution.py

```diff
@@ -1039,12 +1048,86 @@ def getCalibrationSetpoints(self, total_line_scan_time):
+        init_dwellTime = self.parent._ebeam_scanner.dwellTime.value
+        init_physicalFlybackTime = self.physicalFlybackTime.value
+        init_scanOffset = self.scanOffset.value
+        init_scanAmplitude = self.scanAmplitude.value
+        init_cellCompleteResolution = self.parent._mppc.cellCompleteResolution.value
+
+        # Set dwell time, flyback time, and resolution to maximum values to avoid moving the descanners too fast
+        self.parent._ebeam_scanner.dwellTime.value = self.parent._ebeam_scanner.dwellTime.range[-1]
+        self.physicalFlybackTime.value = self.physicalFlybackTime.range[-1]
+        self.parent._mppc.cellCompleteResolution.value = self.parent._mppc.cellCompleteResolution.range[-1]
+        self.scanOffset.value = (0.0, 0.0)
+        scan_amp_rng = self.scanAmplitude.range
+
+        # Set scan amplitude to cover the entire range
+        self.scanAmplitude.value = (scan_amp_rng[0][0], scan_amp_rng[-1][0])
+        total_line_scan_time = self.parent._mppc.getTotalLineScanTime()
+
+        # Generate descanner setpoints for sinusoidal movement in both x and y directions
+        x_descan_setpoints, y_descan_setpoints = self.getCalibrationSetpoints(
+            total_line_scan_time, sine_in_x=True, sine_in_y=True
+        )
+
+        # Generate scanner setpoints (note required by the ASM API, not needed for moving the descanners)
+        x_scan_setpoints, y_scan_setpoints, scan_calibration_dwell_time_ticks = \
+            self.parent._ebeam_scanner.getCalibrationSetpoints(total_line_scan_time)
+
+        # Prepare calibration data
+        calibration_data = CalibrationLoopParameters(self.rotation.value,
+                                                     0,  # Descan X offset parameter unused.
+                                                     x_descan_setpoints,
+                                                     0,  # Descan Y offset parameter unused.
+                                                     y_descan_setpoints,
+                                                     scan_calibration_dwell_time_ticks,
+                                                     self.parent._ebeam_scanner.rotation.value,
+                                                     self.parent._ebeam_scanner.getTicksScanDelay()[0],
+                                                     0.0,  # Scan X offset parameter unused.
+                                                     x_scan_setpoints,
+                                                     0.0,  # Scan Y offset parameter unused.
+                                                     y_scan_setpoints)
+
+        # Check if an acquisition is still in progress
+        if not self.parent._mppc.acq_queue.empty():
+            logging.error("There is still an unfinished acquisition in progress. "
+                          "Calibration mode cannot be started yet.")
+            return
```

Comment:
Does one need to `Restore original parameters and clear calibration parameters` before return?

Reply:
Yes, but I think it actually makes more sense to move this check to the start of this function